### PR TITLE
feat: added middleware.js to protect all pages under admin

### PR DESCRIPTION
**Pull request by ashmeetsmonga**
## Changes proposed

- Added a new file middleware.js in root which handles protected routes (/admin/*) 
- Removed individual files check

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8410"></a>

### CODE REVIEW

File: pages/admin/events.js [hunk 1]
```diff
@@ -16,15 +16,6 @@ import Notification from "@components/Notification";
...
 
-  if (!session) {
-    return {
-      redirect: {
-        destination: "/auth/signin",
-        permanent: false,
-      },
-    };
-  }
-
```
Comment (eddiejaoude) [inline pages/admin/events.js:27]: The middleware should check for `admin` user, please see the lines below...

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution 👍 

Please fix...
- the issue for this middleware is to check for admin users on the path `/admin/*` and `/api/admin/*` with the environment variable `serverEnv.ADMIN_USERS`
- for session only check the path to protect is `/account/*` and `/api/account*`
- admin page `profiles` has been forgotten

**Comment by ashmeetsmonga** [comment]
Nextjs middleware run on Edge runtime, which as far as I have researched, doesn't support env variables. Is there a way to get env variables in middleware?

**Comment by eddiejaoude** [comment]
> I am facing issue with env variables. Nextjs middleware run on Edge runtime, which as far as I have researched, doesn't support env variables. Is there a way to get env variables in middleware?

here are the official Vercel docs https://vercel.com/changelog/more-flexible-environment-variables-in-edge-functions-and-middleware

**Comment by ashmeetsmonga** [comment]
Thank you for your patience and answering my queries. I appreciate it. 
I have few doubts: 
I was getting an error when using `serverEnv.ADMIN_USERS` in middleware.js, so I used `process.env.ADMIN_USERS`.
I have not removed individual check for session in account page, will do that after you approve changes.
Took pull from main after pushing changes (my bad), would remove admin check from profile page also after your approval. 
Thanks

**Comment by eddiejaoude** [comment]
- Did you import for `serverEnv.ADMIN_USERS`? We need to use this to be consistent and also validate the environment variables
- Please remove any commented out code, as this makes it difficult to review and test

**Comment by ChinmayMhatre** [comment]
@eddiejaoude wanted to push this incase it slipped by. @ashmeetsmonga you can post you question in the discord channel as well so that other members in the community can help.

**Pull request closed without merging**